File: Aula16_alunos/app/Http/Controllers/AlunoController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Models\Aluno;
use App\Http\Resources\AlunoResource;
use App\Http\Requests\StoreAlunoRequest;

class AlunoController extends Controller
{
    public function index(){
        return AlunoResource::collection(Aluno::all());
    }
    public function store(StoreAlunoRequest $request){
        $aluno = Aluno::create($request->validated());
        return AlunoResource::make($aluno);
    }
    public function new(){
        return view('alunos.new');
    }
    public function edit($id){
        $aluno = $this->show($id);
        return view('alunos.edit', compact('aluno'));
    }
    public function update(StoreAlunoRequest $request, $id){
        $aluno = Aluno::find($id);
        if($aluno){
            $aluno->update($request->validated());
            return AlunoResource::make($aluno);
        }
        return response()->json(['message'=>'Aluno nao encontrado'], 404);
    }
    public function destroy($id){
        $aluno = Aluno::find($id);
        if($aluno){
            $nome = $aluno->nome;
            $aluno->delete();
            return response()->json(['message'=>'aluno: ' . $nome . ' , foi excluido'], 200);
        }
        return response()->json(null, 204);
    }
    public function show($id){
        $aluno = Aluno::find($id);
        if($aluno)
            return AlunoResource::make($aluno);
        else
            return response()->json(null, 404);
    }
}

File: JavaDesktop_ProjetoAPI/src/Controller/AlunoController.java
package Controller;
import Model.Aluno;
import Service.API;
import java.util.ArrayList;

public class AlunoController {
    public API api = new API();

    public ArrayList<Aluno> index() {
        try {
            return api.index();
        } catch (Exception e) {
            e.getMessage();
            return null;
        }
    }

    public boolean store(Aluno aluno) {
        try {
            Aluno alunoCriado = api.store(aluno);
            return alunoCriado != null;
        } catch (Exception e) {
            e.getMessage();
            return false;
        }
    }

    public boolean update(Aluno aluno) {
        try {
            Aluno alunoEditado = api.update(aluno);
            return alunoEditado != null;
        } catch (Exception e) {
            e.getMessage();
            return false;
        }
    }

    public boolean destroy(int id) {
        try {
            return api.destroy(id);
        } catch (Exception e) {
            e.getMessage();
            return false;
        }
    }
}
